jupyter_sphinx/execute.py

```diff
@@ -351,6 +350,11 @@ def apply(self):
                 [nbformat.v4.new_code_cell(node.astext()) for node in nodes],
                 self.config.jupyter_execute_kwargs,
             )
+
+            for node in nodes:
+                source = node.children[0]
+                source.attributes['language'] = notebook.metadata.language_info.name
```

Comment:
The kernels (ipython most importantly) may also provide a `pygments_lexer` key, which seems better to use here.

e.g. ipython's language is `python`, but `ipython` is a better lexer.

Reply:
good point, will amend now